src/polymer/polymer-element.ts

```diff
@@ -262,9 +262,9 @@ function resolveElement(
     scannedElement: ScannedPolymerElement, document: Document): PolymerElement {
   const element = new PolymerElement();
   element.privacy = scannedElement.privacy;
-  applySuperClass(element, scannedElement, document);
-  applyMixins(element, scannedElement, document);
   applySelf(element, scannedElement, document);
+  applyMixins(element, scannedElement, document);
+  applySuperClass(element, scannedElement, document);
```

Comment:
Can you explain how applying inheritance in order would be O(n^2)?

I disagree that applying in the order is easy to understand, it's backwards from how the language we're modeling does it, so I have to think more about whether it gives the same result.

Behaviors shouldn't be a problem, since Polymer 1 doesn't support inheritance.